### DISCUSSION TITLE
Local build succeeds. Deploy fails

### DISCUSSION THREAD

**Discussion by fotoflo** [vercel/community#4032] [Help] [answered]
### Summary

Cant figure out why this is happening. The build is succeeding on local. then deployment doesnt work 

I dont see any errors.

It also fails when deploying by github push hook

would code outside `/src/` cause build failure without error?

i have a folder which houses some batch processes and one of the APIs is importing functions from that folder - could that cause this?

![image](https://github.com/vercel/community/assets/1642357/e9b652a9-21f9-4618-954a-0ea0c3358a02)

![image](https://github.com/vercel/community/assets/1642357/573c238b-7047-48ff-b859-4ffdc033171b)

package.json
```
{
  "private": true,
  "scripts": {
    "build": "NODE_ENV=production next build",
    "deploy": "vercel deploy",
  },
  "dependencies": {
    "@aws-sdk/client-eventbridge": "^3.406.0",
    "@headlessui/react": "^1.7.10",
    "@heroicons/react": "^2.0.15",
    "@hookform/error-message": "^2.0.1",
    "@hookform/resolvers": "^2.9.10",
    "@next-auth/prisma-adapter": "^1.0.5",
    "@prisma/client": "^4.11.0",
    "@tanstack/react-query": "^4.16.0",
    "@trpc/client": "^10.0.0",
    "@trpc/next": "^10.0.0",
    "@trpc/react-query": "^10.0.0",
    "@trpc/server": "^10.0.0",
    "chalk": "^5.2.0",
    "chart.js": "^4.2.1",
    "chartjs-adapter-date-fns": "^3.0.0",
    "date-fns": "^2.29.3",
    "next": "13.0.2",
    "next-auth": "^4.19.2",
    "next-connect": "^0.13.0",
    "next-superjson-plugin": "^0.5.6",
    "node-fetch": "^2.6.7",
    "node-mailjet": "^5.1.1",
    "nodemailer": "^6.8.0",
    "quirrel": "^1.13.2",
    "react": "18.2.0",
    "react-chartjs-2": "^5.2.0",
    "react-dom": "18.2.0",
    "react-hook-form": "^7.41.5",
    "react-icons": "^4.7.1",
    "styled-components": "^5.3.6",
    "superjson": "1.9.1",
    "uptime-check": "^4.2.0",
    "uuid": "^9.0.0",
    "zod": "3.21.1",
    "zod-error": "^1.5.0",
    "zod-prisma-types": "^2.5.1"
}
```

### Steps to Reproduce

npm run build
npm run deploy

**Comment by amyegan** [comment]
Hey @fotoflo. It could be related to [Build and Development Settings](https://vercel.com/docs/deployments/configure-a-build#build-and-development-settings). If you can share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), that would give us a better view of what might be going wrong.

**Comment by fotoflo** [comment]
hi @amyegan I tried to delete ALL THE THINGS out of my repo and except /pages/api/helloworld.ts, and all the config files - package.json, tsconfig, etc.

that ran build and deploy and it worked.... 

i put them back and stuck on `npm install` again.  

Tried changing the build and development settings but they look ok. 

its fusterating that there's no futher detail on why it's stuck. Going to try "npm install --verbose"

maybe its one of the postinstall scripts from prisma? or is it the thing after these?

... 
```
///  A LOT OF http fetch GET 200s
17:16:19.729 | npm http fetch GET 200 https://registry.npmjs.org/date-fns/-/date-fns-2.29.3.tgz 24266ms (cache miss)
17:16:20.680 | npm http fetch GET 200 https://registry.npmjs.org/@faker-js/faker/-/faker-7.6.0.tgz 25027ms (cache miss)
17:16:20.707 | npm info run prisma@4.16.2 preinstall node_modules/prisma node scripts/preinstall-entry.js
17:16:20.777 | npm info run prisma@4.16.2 preinstall { code: 0, signal: null }
17:16:20.792 | npm info run @prisma/client@4.11.0 postinstall node_modules/@prisma/client node scripts/postinstall.js
17:16:20.807 | npm info run @prisma/engines@4.16.2 postinstall node_modules/@prisma/engines node scripts/postinstall.js
17:16:20.822 | npm info run styled-components@5.3.6 postinstall node_modules/styled-components node ./postinstall.js
17:16:20.910 | npm info run styled-components@5.3.6 postinstall { code: 0, signal: null }
17:16:22.086 | npm info run @prisma/engines@4.16.2 postinstall { code: 0, signal: null }
````


![image](https://github.com/vercel/community/assets/1642357/d99c7ded-a1fa-43cc-b1aa-a6f0b8986686)

**Reply by amyegan** [reply]
You could also try using the [CLI](https://vercel.com/docs/cli). The `vercel build --debug` command will kick off a local build that writes extra info

**Comment by eliawk** [comment]
hello @amyegan;
I've the same problem: build on local and on vercel ar working but then the deploy fails:
`

Built in 9.5s
--
10:41:52.126 | Done in 10.69s.
10:42:00.694 | Build Completed in /vercel/output [2m]
10:42:01.032 | Deploying outputs...
10:42:02.930 | An unexpected error happened when running this build. We have been notified of the problem. If you have any questions, please contact Vercel Support https://vercel.com/help.
10:42:02.959 | Injected preview comments middleware, you can disable this in your project settings.

`
any ideas how to fix that?

**Reply by charlottedamm** [reply]
Did you ever manage to fix it or find out what's wrong? Having the same exact issue

**Reply by sannajammeh** [reply]
Having the exact same issue

**Reply by amyegan** [reply]
Do you get more error information if you using the [CLI](https://vercel.com/docs/cli) and run `vercel build` or `vercel build --debug` in your local repo?

**Reply by sannajammeh** [reply]
No error information at all @amyegan . The entire project builds perfectly fine, runs perfectly fine locally, but Vercel hangs for 30 minutes attempting to deploy the outputs. It fails 9/10 times and some cases go through. We have gone through the entire commit log going back to the days it worked and see no reason why it would fail. 

We have an open ticket: #00202360

**Reply by amyegan** [reply]
Thanks for sharing that. It looks like you're already working with a customer success engineer and they requested a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) to help with investigation into this error. If you're able to share a simplified example with them, it will help narrow down the list of possible causes.

I also thought of one more relatively quick thing to try. Since you said the Vercel CLI build succeeded locally, could you try [deploying from local build](https://vercel.com/docs/cli/deploying-from-cli#deploying-from-local-build-prebuilt)? You can try a prebuilt preview deployment by running `vercel build` and then `vercel deploy --prebuilt`. Please let me (or your support engineer) know if that works or share the output if it fails. 🙏

**Reply by sannajammeh** [reply]
I'm afraid we are unable to reproduce the issue, we've been trying for 2 days flipping different configuration options. Its happening at random. We just had another deployment work just fine and a single line code change in the same branch broke the next deployment again.

If we re-deploy the one that works it will also fail. Its completely random.

`vercel deploy --prebuilt` also fails with the same output:

```bash
Error: An unexpected error happened when running this build. We have been notified of the problem. If you have any questions, please contact Vercel Support https://vercel.com/help.
    at Now.handleDeploymentError (/Users/sanna/Library/pnpm/global/5/.pnpm/vercel@33.5.2/node_modules/vercel/dist/index.js:165689:16)
    at processDeployment (/Users/sanna/Library/pnpm/global/5/.pnpm/vercel@33.5.2/node_modules/vercel/dist/index.js:165449:34)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Now.create (/Users/sanna/Library/pnpm/global/5/.pnpm/vercel@33.5.2/node_modules/vercel/dist/index.js:165578:28)
    at async createDeploy (/Users/sanna/Library/pnpm/global/5/.pnpm/vercel@33.5.2/node_modules/vercel/dist/index.js:165175:12)
    at async deploy_default (/Users/sanna/Library/pnpm/global/5/.pnpm/vercel@33.5.2/node_modules/vercel/dist/index.js:172716:22)
    at async main13 (/Users/sanna/Library/pnpm/global/5/.pnpm/vercel@33.5.2/node_modules/vercel/dist/index.js:199254:18)
Error: An unexpected error happened when running this build. We have been notified of the problem. If you have any questions, please contact Vercel Support https://vercel.com/help.
```

**Reply by charlottedamm** [reply, marked as answer]
When we were having the issue we were told to add the env variable "VERCEL_ENABLE_EXTENDED_FALLBACK_PAYLOAD" with the value 1, and that fixed our builds.

**Reply by sannajammeh** [reply]
You're a lifesaver @charlottedamm !!! It works across all branches now 🚀

**Reply by frans-de-waal** [reply]
@charlottedamm Do you know if there is any documentation regarding the VERCEL_ENABLE_EXTENDED_FALLBACK_PAYLOAD option? Google only brings up this Github issue.

**Reply by amyegan** [reply]
There is no official documentation for `VERCEL_ENABLE_EXTENDED_FALLBACK_PAYLOAD` because it is not a stable feature. It is a temporary flag to enable an updated strategy for handling dynamic tags in requests. 

The environment variable is a way to opt yourself in early for this experimental feature. It will be enabled for everyone after testing is completed to ensure nothing else is unintentionally affected, and you will be able to delete the env var then.

**Comment by eliawk** [comment]
I'm using a custom routes for Remix.js  - https://remix.run/docs/en/main/discussion/routes#manual-route-configuration
